### DISCUSSION TITLE
Question: How  POS consensus for polygon edge works?

### DISCUSSION THREAD

**Discussion by Kvenessa** [0xPolygon/polygon-edge#458] [Q&A] [answered]
1. What is the selection process followed to select a validator in an epoch? 
2. How stakers vote for the validators?
3. Why it is called IBFT POS and IBFT POA?

**Comment by Kourin1996** [comment, marked as answer]
@Kvenessa 

Please note our PoS is under development. There would be some functions to be added into PoS.
Currently, Staking contract maintains a validator set and an account that has staked enough amount into the contract will be a validator. The validator set will not update in the consensus layer after immediately the validator set in the contract changed. The consensus layer fetches validator set from the Staking contract every epoch (every 100000 blocks as default) and set it to own validator set.

then I'd like to answer your questions:
1. Currently it's same to PoA's one. Block producer changes every block or round and the block producer is selected in order from the current validator set.
2. No delegation mechanism at the moment. The staker that has staked enough amount will be validator from the next epoch. Currently, there's no function for the staker that has not staked enough amount.
3. Because it's using IBFT as consensus. IBFT defines how and who produces a new block, how validators verify, how to get consensus, etc. On the other hand, PoS will define who becomes a block producer at a height.
